### PR TITLE
Update golangci-lint v1.35.2 -> v1.37.0

### PR DESCRIPTION
**Pull request by 6543**
as title

### CODE REVIEW

**Review by jolheiser — APPROVED** [review]
Keep in mind this won't do anything for users who already have the binary unless they remember to update manually.

**Comment by 6543** [comment]
@jolheiser see discord #maintainer chat ... - we have more and more issues in CI with golangci-lint and this looks like it fix it

**Review by a1012112796 — APPROVED** [review]

**Comment by a1012112796** [comment]
I'd like pre install it in [test-env](https://www.gitea.com/gitea/test-env) also.

File: Makefile [hunk 1]
```diff
@@ -708,7 +708,7 @@ pr\#%: clean-all
 golangci-lint:
 	@hash golangci-lint > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
```
Comment (a1012112796) [inline Makefile:709]: wonder if we need request min version..

**Comment by 6543** [comment]
@a1012112796 think this is worth it's own pull ... merge this fast so CI passes more often again ...

**Pull request merged**